python_modules/dagster-graphql/dagster_graphql/schema/asset_graph.py

```diff
@@ -235,6 +235,7 @@ class GrapheneAssetNode(graphene.ObjectType):
     backfillPolicy = graphene.Field(GrapheneBackfillPolicy)
     computeKind = graphene.String()
     configField = graphene.Field(GrapheneConfigTypeField)
+    cycle = graphene.List(graphene.NonNull(GrapheneAssetKey))
```

Comment:
hmm, what other graphql schema did you consider for this? per asset cycle list seems a little odd to me on first consideration

Reply:
If by cycle list you mean "list of cycles", that's not what this is. This is either null or a single list of asset keys per asset. If it's a list of asset keys, that represents a strongly connected component (which contains at least one but possibly more) cycles containing the asset. Each node in a graph is part of at most one strongly connected component.

I think that we need to deliver this info somehow so it can be shown in the UI, though maybe it would make sense to hoist it to the code location level. The primary reason I did it on `AssetNode` is that it was easier to slot into existing query/resolver logic.

Reply:
> maybe it would make sense to hoist it to the code location level. The primary reason I did it on AssetNode is that it was easier to slot into existing query/resolver logic.

ya hoisting it above was what i imagined might be easier for the frontend but would defer to @bengotow on making that call

Reply:
Hey gang sorry for the delay on this -- I do think it would be nice to hoist to the repo level or a separate top level resolver so that we don't have to fetch this field on each asset node.

More broadly though I'm concerned about the granularity of this approach. 

1) What is the UI supposed to do to `handle the situation appropriately` after loading this information?
- My guess is that we'll forcibly remove one of the offending edges to break the cycle, or just not render the asset graph containing these assets at all. If that's the case, I would prefer to actually break the cycle in the backend asset graph representation at load-time and tell the user we did it in the UI.

2) If the asset graph DID load successfully, would the rest of the app behave correctly? Could you launch a run, etc? My guess is that there are other problems, on both the frontend and backend (stale cause traversal?) that we'll hit after we address #1.

I think it'd be preferable for the load of the second code location to completely error with a message saying that you need to resolve your graph cycles and try again. I feel like cycles are a core assertion failure for the entire platform and not just a "hide that edge in the graph and continue" scenario?

If I'm wrong about #2 I'm happy to do the front-end parts for #1 of course! Just worried we'll be finding little bugs related to cycles in other parts of the product for a little while

Reply:
Hey gang sorry for the delay on this -- I do think it would be nice to hoist this new resolver to the repo level or a separate top level resolver so that we don't have to fetch this field on each asset node.

I think you're right @smackesey that we can improve the existing cycle detection escape logic in the UI using this resolver as a first pass 👍 

I still have reservations about allowing users to try to use the platform as a whole with cycles in their graphs and worry this is just the first and most obvious problem, but this is definitely a big improvement over our current handling!